### PR TITLE
WIP:  Add preliminary command line options to main.cpp

### PR DESCRIPTION
**Pull request by cmhobbs**
Just a RFC here.  I'm not ready to have this merged but I'm trying to tackle issue #219.  Note that this is my first dive into C++ so this is probably not very pretty but I've tried to stick with the style and contributing guidelines.

I've added the debug switches and multiplayer settings.  I have not yet added the `-v`/`--version` or the `--data-dir` path.  I haven't even hunted for those yet and I don't know if they currently exist.  I wanted to make sure I'm headed down the right path here.

Thanks for your patience and thanks for devilutionX! :smiling_imp:

### CODE REVIEW

File: SourceX/main.cpp [hunk 1]
```diff
@@ -19,13 +19,49 @@ static std::string build_cmdline(int argc, char **argv)
...
+	// TODO (cmhobbs)
+	// - [X] catch -h and --help in cmdline()
+	// - [X] add debug output options
+	//   - ref: https://github.com/diasurgical/devilutionX/blob/master/docs/debug.md
+	// - [ ] add options from GHI
+	//   - ref: https://github.com/diasurgical/devilutionX/issues/219
+	//   - [ ] -h and --help to print help options
+	//   - [ ] -v and --version to show version or git branch/rev/commit
+	//   - [ ] --data-dir path to diabdat.mpq
+	if ((cmdline == "-h") || (cmdline == "--help")) {
```
Comment (cmhobbs) [inline SourceX/main.cpp]: Maybe it makes more sense to stuff all this into its own function?

**Comment by cmhobbs** [comment]
I did also consider using Boost.Program_options as it's used in a 3rd party library but that seemed overkill, plus it doesn't look like boost is being used by anything else at the moment.

**Comment by MBeijer** [comment]
Please don't use boost unless necessary. It's such unnecessary bloat.

**Comment by cmhobbs** [comment]
Duly noted, thanks.  Again, I'm just getting started with C++ so I don't know what's canonical.  I didn't use it here because I only saw it in one part of the project and that was in the 3rd party directory.  I know very little about boost beyond the strong feelings it seems to evoke. :smile:

**Comment by MBeijer** [comment]
Boost has its uses, but there is a tendency to include it because of laziness. The most useful parts are being integrated in the newer C++ standards too, so there's less reason to use the library.

File: SourceX/main.cpp (hunk 1 above)
Comment (qndel) [reply SourceX/main.cpp]: please choose one, sometimes you write "enable", sometimes "enables" lol :P

Comment (cmhobbs) [reply SourceX/main.cpp]: I took the descriptions directly from docs/debug.md.  Which do we prefer?

Comment (qndel) [reply SourceX/main.cpp]: I'd go with "enable" because it's shorter 🤣

Comment (cmhobbs) [reply SourceX/main.cpp]: Great!  I'll make the changes.  Thanks for the input!

**Comment by cmhobbs** [comment]
Got plenty of input here.  Going to close this PR and open a new one with changes suggested and the rest of issue #291.  Thanks again!

**Pull request closed without merging**